vault/resource_aws_auth_backend_client.go

```diff
@@ -79,79 +80,105 @@ func awsAuthBackendClientResource() *schema.Resource {
+	stsRegion := d.Get(consts.FieldSTSRegion).(string)
+	stsRegionFromClient := d.Get(useSTSRegionFromClient).(bool)
+
+	identityTokenAud := d.Get(consts.FieldIdentityTokenAudience).(string)
+	roleArn := d.Get(consts.FieldRoleArn).(string)
+	identityTokenTTL := d.Get(consts.FieldIdentityTokenTTL).(int)
 
-	iamServerIDHeaderValue := d.Get("iam_server_id_header_value").(string)
+	iamServerIDHeaderValue := d.Get(consts.FieldIAMServerIDHeaderValue).(string)
 
 	path := awsAuthBackendClientPath(backend)
 
 	data := map[string]interface{}{
-		"endpoint":                   ec2Endpoint,
-		"iam_endpoint":               iamEndpoint,
-		"sts_endpoint":               stsEndpoint,
-		"sts_region":                 stsRegion,
-		"iam_server_id_header_value": iamServerIDHeaderValue,
+		"endpoint":                         ec2Endpoint,
+		consts.FieldIAMEndpoint:            iamEndpoint,
+		consts.FieldSTSEndpoint:            stsEndpoint,
+		consts.FieldSTSRegion:              stsRegion,
+		consts.FieldIAMServerIDHeaderValue: iamServerIDHeaderValue,
 	}
 
-	if d.HasChange("access_key") || d.HasChange("secret_key") {
+	if d.HasChange(consts.FieldAccessKey) || d.HasChange(consts.FieldSecretKey) {
 		log.Printf("[DEBUG] Updating AWS credentials at %q", path)
-		data["access_key"] = d.Get("access_key").(string)
-		data["secret_key"] = d.Get("secret_key").(string)
+		data[consts.FieldAccessKey] = d.Get(consts.FieldAccessKey).(string)
+		data[consts.FieldSecretKey] = d.Get(consts.FieldSecretKey).(string)
 	}
 
 	if provider.IsAPISupported(meta, provider.VaultVersion115) {
 		data[useSTSRegionFromClient] = stsRegionFromClient
 	}
 
+	if provider.IsAPISupported(meta, provider.VaultVersion117Ent) {
+		data[consts.FieldIdentityTokenAudience] = identityTokenAud
+		data[consts.FieldRoleArn] = roleArn
+		data[consts.FieldIdentityTokenTTL] = identityTokenTTL
+	}
+
 	// sts_endpoint and sts_region are required to be set together
 	if (stsEndpoint == "") != (stsRegion == "") {
-		return fmt.Errorf("both sts_endpoint and sts_region need to be set")
+		return diag.Errorf("both sts_endpoint and sts_region need to be set")
 	}
 
 	log.Printf("[DEBUG] Writing AWS auth backend client config to %q", path)
 	_, err := client.Logical().Write(path, data)
 	if err != nil {
-		return fmt.Errorf("error writing to %q: %s", path, err)
+		return diag.Errorf("error writing to %q: %s", path, err)
 	}
 	log.Printf("[DEBUG] Wrote AWS auth backend client config to %q", path)
 
 	d.SetId(path)
 
-	return awsAuthBackendRead(d, meta)
+	return awsAuthBackendRead(ctx, d, meta)
 }
 
-func awsAuthBackendRead(d *schema.ResourceData, meta interface{}) error {
+func awsAuthBackendRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client, e := provider.GetClient(d, meta)
 	if e != nil {
-		return e
+		return diag.FromErr(e)
 	}
 
 	log.Printf("[DEBUG] Reading AWS auth backend client config")
 	secret, err := client.Logical().Read(d.Id())
```

Comment:
nit: May as well use the context since we have it
```suggestion
	secret, err := client.Logical().ReadWithContext(ctx, d.Id())
```